platforms/software/ivy/src/main/java/org/gradle/api/publish/ivy/internal/artifact/FileBasedIvyArtifact.java

```diff
@@ -17,48 +17,57 @@
 package org.gradle.api.publish.ivy.internal.artifact;
 
 import com.google.common.io.Files;
+import org.gradle.api.file.RegularFile;
+import org.gradle.api.internal.provider.Providers;
 import org.gradle.api.internal.tasks.TaskDependencyFactory;
 import org.gradle.api.internal.tasks.TaskDependencyInternal;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.provider.ProviderFactory;
 import org.gradle.api.publish.ivy.internal.publisher.IvyPublicationCoordinates;
 import org.gradle.api.tasks.TaskDependency;
 
 import java.io.File;
 
 public class FileBasedIvyArtifact extends AbstractIvyArtifact {
-    private final File file;
-    private final String extension;
+    private final Provider<RegularFile> file;
```

Comment:
Seems odd we're storing a `Provider` here if we always have a non-lazy file passed in the constructor